src/PrestaShopBundle/Routing/Converter/ResetCacheRouterProvider.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Routing\Converter;
+
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+class ResetCacheRouterProvider extends AbstractLegacyRouteProvider
```

Comment:
The purpose of this class is a bit foggy, it's used both to clear the cache, but also generate the cache right away It's kind of mixing two purposes Even the way it's used, it's only used in `FeatureFlagsFormDataProvider` and it calls `getLegacyRoutes` without caring about the result just to force the cache to be cleared.

I think this should be changed into a dedicated `LegacyRouteCacheCleaner` with a single `clearCache` method, this way its responsibility is unique and well identified. The cache will automatically be restored in the next call by the cache provider anyway.

Reply:
There is no way to remove the cache, you can only set a value to override. You can set the cache value to null, but we should then edit the cache provider to check null values.

Reply:
Are you sure you can't use `$this->cache->deleteItem($this->cacheKeyGenerator->getCacheKey())` ?